### PR TITLE
fix(Core/Player): Send player's own auras before all visible objects

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/c6cd272505cdca5d21935ba346df82b8c767fb13)

## Changes Proposed:
-  This will fix the very rare bug of characters committing indecent exposure. Which is very illegal should you know.
( Rare bug where some players appear naked for individual players)

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -11958,9 +11957,9 @@ void Player::learnSkillRewardedSpells(uint32 skill_id, uint32 skill_value)
     }
 }
 
-void Player::GetAurasForTarget(Unit* target) // pussywizard: contact before changing ANYTHING!
+void Player::GetAurasForTarget(Unit* target, bool force /*= false*/)
 {
-    if (!target/* || target->GetVisibleAuras()->empty()*/)                  // speedup things
+    if (!target || (!force && target->GetVisibleAuras().empty())    // speedup things
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.cpp:11962]:
```suggestion
    if (!target || (!force && target->GetVisibleAuras()->empty())    // speedup things
```

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -11958,9 +11957,9 @@ void Player::learnSkillRewardedSpells(uint32 skill_id, uint32 skill_value)
     }
 }
 
-void Player::GetAurasForTarget(Unit* target) // pussywizard: contact before changing ANYTHING!
+void Player::GetAurasForTarget(Unit* target, bool force /*= false*/)
 {
-    if (!target/* || target->GetVisibleAuras()->empty()*/)                  // speedup things
+    if (!target || (!force && target->GetVisibleAuras()->empty())    // speedup things
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.cpp:11962]:
```suggestion
    if (!target || (!force && target->GetVisibleAuras()->empty()))    // speedup things
```

File: src/server/game/Maps/Map.cpp [hunk 3]
```diff
@@ -2534,15 +2535,25 @@ void Map::SendInitSelf(Player* player)
...
+    // build and send self update packet before sending to player his own auras
+    data.BuildPacket(&packet);
+    player->SendDirectMessage(&packet);
+
+    // send to player his own auras (this is needed here for timely initialization of some fields on client)
+    player->GetAurasForTarget(player, true);
+
+    // clean buffers for further work
+    packet.clear();
+    data.clear();
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.cpp:2547]:
```suggestion
    data.Clear();
```

**Comment by Kofmy** [comment]
degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates degenerates

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**